include/LightGBM/config.h

```diff
@@ -505,6 +505,13 @@ struct Config {
   // descl2 = note that the parent output ``w_p`` itself has smoothing applied, unless it is the root node, so that the smoothing effect accumulates with the tree depth
   double path_smooth = 0;
 
+  // desc = controls which features can appear in the same branch
+  // desc = for CLI, specify as lists separated by commas, e.g. ``[0,1,2],[2,3]``
+  // desc = for Python, specify as list of lists, e.g. ``[[0, 1, 2], [2, 3]]``
+  // desc = not yet supported for R
+  // desc = any two features can only appear in the same branch only if there exists a constraint containing both feaures
```

Comment:
I think we should clarify, that this is disabled by default. Something like 
```
// desc = by default interaction constraints are disabled, to enable them you can specify
// descl2 = for CLI, lists separated by commas, ...
// descl2 = for Python-package, list of lists, ...
// descl2 = for R-package, **not yet supported**
```

```suggestion
  // desc = any two features can only appear in the same branch only if there exists a constraint containing both features
```